### PR TITLE
[Doc] Update Iceberg catalog

### PR DESCRIPTION
**Pull request by amber-create**
Signed-off-by: amber-create <yangyanping@starrocks.com>

Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: docs/data_source/catalog/iceberg_catalog.md [hunk 1]
```diff
@@ -10,12 +10,12 @@ To ensure successful SQL workloads on your Iceberg cluster, your StarRocks clust
...
 - Distributed file system (HDFS) or object storage like AWS S3, Microsoft Azure Storage, Google GCS, or other S3-compatible storage system (for example, MinIO)
 
-- Metastore like Hive metastore, AWS Glue, or REST
+- Metastore like Hive metastore, AWS Glue, or Tabular
 
   > **NOTE**
   >
   > - If you choose AWS S3 as storage, you can use HMS or AWS Glue as metastore. If you choose any other storage system, you can only use HMS as metastore.
-  > - If you use Tabular Iceberg Catalog, you must choose REST as metastore.
+  > - If you choose Tabular as metastore, you need to use the Iceberg REST catalog.
```
Comment (evelynzhaojie) [inline docs/data_source/catalog/iceberg_catalog.md:18]: maybe change "choose" to "user" for consistent word choice. Row 171 and 209 use "use".

**Review by evelynzhaojie — APPROVED** [review]

**Review by EsoragotoSpirit — APPROVED** [review]

**Comment by amber-create** [comment]
@mergify backport branch-3.1

**Pull request merged**